llm/kv_cache.go

```diff
@@ -0,0 +1,102 @@
+package llm
+
+import (
+	"fmt"
+	"log/slog"
+	"slices"
+
+	"github.com/ollama/ollama/discover"
+	"github.com/ollama/ollama/envconfig"
+)
+
+// Interface for GGML functionality needed by validation
+type GGMLModel interface {
+	KV() KV
+}
+
+// ModelSupportsFlashAttention checks if the model supports flash attention
+func ModelSupportsFlashAttention(ggml GGMLModel) bool {
```

Comment:
Similar to below, this can be lowercased
```suggestion
func modelSupportsFlashAttention(ggml GGMLModel) bool {
```